File: 2610.convert-an-array-into-a-2-d-array-with-conditions.cs
/*
 * @lc app=leetcode id=2610 lang=csharp
 *
 * [2610] Convert an Array Into a 2D Array With Conditions
 */

// @lc code=start
using System;
using System.Collections.Generic;

public class Solution {
    public IList<IList<int>> FindMatrix(int[] nums) {
        Array.Sort(nums);
        List<IList<int>> result = new List<IList<int>>();
        int prev = 0;
        int j = 0;
        
        foreach (int num in nums) {
            if (num != prev) {
                j = 0;
            }

            if (0 <= j && j < result.Count) {
                result[j].Add(num);
            }
            else {
                result.Add(new List<int> { num });
            }

            j++;
            prev = num;
        }

        return result;
    }
}

// @lc code=end

File: 2610.convert-an-array-into-a-2-d-array-with-conditions.py
#
# @lc app=leetcode id=2610 lang=python3
#
# [2610] Convert an Array Into a 2D Array With Conditions
#

# @lc code=start
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        result = []
        prev = 0
        for num in nums:
            if num != prev: j = 0
            if 0 <= j < len(result):
                result[j].append(num)
            else:
                result.append([num])
            j += 1
            prev = num
        return result
# @lc code=end

